packages/repo/src/storage/memory-blockstore.ts

```diff
@@ -1,43 +1,57 @@
 import { CID } from 'multiformats/cid'
-import RepoStorage from './repo-storage'
-import { CommitBlockData, CommitData, def } from '../types'
+import { CommitData, def } from '../types'
 import BlockMap from '../block-map'
 import { MST } from '../mst'
+import DataDiff from '../data-diff'
+import { MissingCommitBlocksError } from '../error'
+import RepoStorage from './repo-storage'
 
 export class MemoryBlockstore extends RepoStorage {
   blocks: BlockMap
   head: CID | null = null
 
-  constructor() {
+  constructor(blocks?: BlockMap) {
     super()
-    this.blocks = new BlockMap()
+    this.blocks = blocks || new BlockMap()
```

Comment:
Would there be any value in cloning `blocks` to avoid leakage?  I could see it arguably the other way too, that it's useful for testing purposes to mutate the passed block map.

Reply:
hmm that's a good point. Yeah I like cloning 👍